File: src/main/java/ai/datahunters/md/server/photos/search/json/SearchResponse.java
package ai.datahunters.md.server.photos.search.json;

import ai.datahunters.md.server.photos.search.json.response.FacetFieldResult;
import lombok.Builder;
import lombok.Value;

import java.util.List;
import java.util.Map;

@Value
@Builder
public class SearchResponse {
    private List<Photo> photos;
    private Map<String, FacetFieldResult> facets;
    private Integer page;
    private Long total;
}

File: core/src/main/scala/ai/datahunters/md/server/photos/search/SearchResponse.scala
package ai.datahunters.md.server.photos.search

case class SearchResponse(
    photos: List[PhotoEntity],
    facets: Map[Field, Map[String, Long]],
    page: Int,
    total: Long,
    possibleFilters: Set[PossibleFilter])

case class Dupa(map: Map[Field, Map[String, Long]])
